fastparse/shared/src/main/scala/fastparse/ParserApi.scala

```diff
@@ -24,17 +24,20 @@ trait ParserApi[+T, ElemType, Repr] {
   /**
    * Repeats this parser 0 or more times
    */
-  def rep[R](implicit ev: Repeater[T, R]): Parser[R, ElemType, Repr]
+  def rep[R](implicit ev: Repeater[T, R],
+             formatter: ElemTypeFormatter[ElemType]): Parser[R, ElemType, Repr]
```

Comment:
Can the `formatter` be part of the `ParserInput` object, rather than being passed in anywhere as an implicit? I assume it won't change between the various parsers run on the same input, so it would make sense to have it specified once rather than specified all over the place